### ISSUE TITLE
write run_election

### ISSUE THREAD

**Issue by jamesturk** [mggg/VoteKit#14]
this depends upon #11 at least, but should be pretty straightforward once it's there

right now the imagined signature is something like:

```
def run_election(cleaned_pp: PreferenceProfile, election_step: Callable[[PreferenceProfile], [(PreferenceProfile, StepOutcome)]) -> ElectionOutcome
```

But perhaps it'll wind up wrapped in a different interface/class.

The key here is that the StepOutcomes are coalesced into an ElectionOutcome, which are currently presumed to have different needs.

**Comment by jamesturk** [comment]
@jgibson517, wanted to write down some thoughts on the iterable approach you had in mind & some options.

As I mentioned, the easiest way to get there would be to use generator functions.  Here's a psuedocode version:

```
def iter_election(pp: PreferenceProfile, step_function: StepFunc):
    while True:
        pp, result = step_function(pp)
        yield result
        # aggregate / use result as needed here
        if result.is_finished():
            break
```

This function returns a generator, which is a type of iterable where every `__next__` calls the function again, until the function exits normally.

Usage looks like this:

```
for intermediate_result in iter_election(...):
      # this loop will execute every time the yield condition in the above code is met
      print(intermediate_result)

# You could also get them all at once by casting to a list:
list(iter_election(...))
```

You could then use this method to implement a `run_election` method for convinience.  It might wind up not being much more than a call to list(iter_election()) but is likely a nice-to-have for users.
